src/backend/data/feed.js

```diff
@@ -37,7 +44,34 @@ class Feed {
    * Returns a Promise.
    */
   save() {
-    addFeed(this);
+    return addFeed(this);
+  }
+
+  /**
+   * Removes current Feed + associated Posts from the databases.
+   * Returns a Promise
+   */
+  async delete() {
+    await removeFeed(this.id);
+    const posts = await getPosts(0, 0);
+    const filteredPosts = posts.filter(async (post) => (await getPost(post).feed) === this.id);
+    const expandedPosts = await Promise.all(
+      await filteredPosts.map((filteredPost) => getPost(filteredPost))
```

Comment:
I don't think you need to `await` here, you should be returning an Array of Promises.